src/Plugin/GraphQL/DataProducer/Entity/EntityQuery.php

```diff
@@ -0,0 +1,166 @@
+ *       label = @Translation("Offset"),
+ *       required = FALSE
+ *     ),
+ *     "conditions" = @ContextDefinition("any",
+ *       label = @Translation("Conditions"),
+ *       multiple = TRUE,
+ *       required = FALSE
+ *     ),
+ *     "language" = @ContextDefinition("string",
+ *       label = @Translation("Entity languages(s)"),
+ *       multiple = TRUE,
+ *       required = FALSE
+ *     ),
+ *     "bundles" = @ContextDefinition("any",
+ *       label = @Translation("Entity bundle(s)"),
+ *       multiple = TRUE,
+ *       required = FALSE
+ *     )
+ *   }
+ * )
+ */
+class EntityQuery extends DataProducerPluginBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * Maximum number of results.
+   *
+   * To prevent denial of service attacks with loading too many items.
+   */
+  const SIZE_MAX = 50;
+
+  /**
+   * The entity type manager service.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManager
+   */
+  protected $entityTypeManager;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * EntityLoad constructor.
+   *
+   * @param array $configuration
+   *   The plugin configuration array.
+   * @param string $pluginId
+   *   The plugin id.
+   * @param array $pluginDefinition
+   *   The plugin definition array.
+   * @param \Drupal\Core\Entity\EntityTypeManager $entityTypeManager
+   *   The entity type manager service.
+   */
+  public function __construct(
+    array $configuration,
+    string $pluginId,
+    array $pluginDefinition,
+    EntityTypeManager $entityTypeManager
+  ) {
+    parent::__construct($configuration, $pluginId, $pluginDefinition);
+    $this->entityTypeManager = $entityTypeManager;
+  }
+
+  /**
+   * Resolves the entity query.
+   *
+   * @param string $type
+   *   Entity type.
+   * @param int $limit
+   *   Maximum number of queried entities.
+   * @param int|null $offset
+   *   Offset to start with.
+   * @param array|null $conditions
+   *   List of conditions to filter the entities.
+   * @param string|null $language
+   *   Language of queried entities.
+   * @param array|null $bundles
+   *   List of bundles to be filtered.
+   * @param \Drupal\Core\Cache\RefinableCacheableDependencyInterface $metadata
+   *   The metadata object for caching.
+   *
+   * @return array
+   *   The list of ids that match this query.
+   *
+   * @throws \GraphQL\Error\UserError
+   *   No bundles defined for given entity type.
+   */
+  public function resolve(string $type, int $limit = 10, ?int $offset = NULL, ?array $conditions = NULL, ?string $language = NULL, ?array $bundles = NULL, RefinableCacheableDependencyInterface $metadata): array {
+
+    // Make sure that max limit is not crossed.
+    if ($limit > static::SIZE_MAX) {
```

Comment:
Let's remove that limit. It's up to the implementation to make sure it's not above a certain threshold.